tools/trap/trap.c

```diff
@@ -463,6 +528,40 @@ int main(int argc, char *argv[])
 			printf("\nExternal Userfs partition dump received successfully\n");
 			fflush(stdout);
 		}
+
+		if (DUMP_FLAGS & FILE_DUMP_FLAG) {
+#ifdef CONFIG_SMARTFS_MAXNAMLEN
+			int smartfs_maxnamelength = CONFIG_SMARTFS_MAXNAMLEN;
+#else
+			int smartfs_maxnamelength;
+			printf("No Config found for maximum possible file name length, please enter manually (usually = 32)\n");
+			printf("You may ignore any config.h header file inclusion errors in this case\n");
+			scanf("%d\n", smartfs_maxnamelength);
+#endif
+			char filename[CONFIG_SMARTFS_MAXNAMLEN];
+			printf("This utility will allow you to enter a file name that you want to be dumped from the device\n");
+			printf("It is the responsibility of the user to ensure that the correct file name/path is provided\n");
+
+			strncpy(handshake_string, USERFS_FDUMP_HANDSHAKE_STRING, HANDSHAKE_STR_LEN_MAX);
+			if (do_handshake(dev_fd, handshake_string) != 0) {
+				printf("Target handshake failed\n");
+				goto handshake_err;
+			}
+
+			printf("Enter the path of the file that you want to be dumped from the Filesystem: -");
```

Comment:
Is it **absolute path** or path relative to **/mnt**  ? I think its better to mention here to be clear.

Reply:
Let's mention absolute